### ISSUE TITLE
Remove locale redirects

### ISSUE THREAD

**Issue by pmac** [mozilla/bedrock#8862]
Currently the server will intercept requests to pages without a locale in the URL, figure out the best translation of the page based on available translations and the user's `Accept-Language` header, and redirect them to the best translation. This has negative effects on SEO in that crawlers don't like it as much and it makes the time from initial request to showing content longer. It also means far more requests to our server infrastructure since these requests aren't very easily cached by our CDN because it can be different for every user.

## Proposal

We should have URLs that have no language in them render the page in the default language, which for this site is `en-US`. Once #6857 is done if the user lands on this locale-free URL but have their browser set to indicate that they prefer another language for which we have a translation, the translation bar would pop up and inform them of this and make it easy for them to switch to their preferred language.

This method would make the server-side of the site more simple, easier to run, easier to cache and thus cheaper to run, and would do a better job to ensure that the user is on the best version of each page for them. We would have a redirect of every `/en-US/...` URL to the equivalent `/...` URL but these redirects would not depend on the user's individual language config and would therefore be cacheable at the CDN and therefore avoid the same issue we have now.

This would also give us more flexibility in deployment strategies in that we could deploy most of the site's HTML as static files if we needed or wanted to for site reliability or cost reasons.

**Comment by rraue** [comment]
So just to make sure I completely understand your suggestion: We would not have /en-US/xxx anymore but all that content would be redirected to /xxx, right? 

Would the translation bar appear always on top of every page or just if the user is redirected to /xxx?

We should test this for UX and conversions. Another way to tackle this would be duplicating /en-US/xxx content on /xxx with an additional big translation bar on top of that content, canonicalize it to /en-US/ so we do not get a duplicate content problem but leave /en-US/ untouched as it is. A user visiting /xxx would get the same experience and good possibility to change language but a "direct" visitor to en-US/xxx would get still the same experience as now. And both would be cacheable, if I am not mistaken. Not saying this is the better solution if any, but maybe another way to have less conditions and aim for the same goal without even risking any conversions.

**Comment by pmac** [comment]
> So just to make sure I completely understand your suggestion: We would not have /en-US/xxx anymore but all that content would be redirected to /xxx, right?

Right

> Would the translation bar appear always on top of every page or just if the user is redirected to /xxx?

The translation bar looks at `navigator.languages` to see if the current translation is one of the user's preferred languages. If the page is available in another translation we have the bar will pop up in the language we think they prefer and offer to take them to the translated version. See #6857 for more info on the bar.
 
> We should test this for UX and conversions. Another way to tackle this would be duplicating /en-US/xxx content on /xxx with an additional big translation bar on top of that content, canonicalize it to /en-US/ so we do not get a duplicate content problem but leave /en-US/ untouched as it is. A user visiting /xxx would get the same experience and good possibility to change language but a "direct" visitor to en-US/xxx would get still the same experience as now. And both would be cacheable, if I am not mistaken. Not saying this is the better solution if any, but maybe another way to have less conditions and aim for the same goal without even risking any conversions.

That's fine with me. I was just figuring that having the same content at 2 urls was undesirable, but if serving it instead of redirecting is just as good then I agree that we should just do that.

**Comment by rraue** [comment]
I will have to think about it whats the "best" for SEO, as both have possible pros and cons, but none seems like a blocker to me.

**Comment by maureenlholland** [comment]
I believe this is a duplicate of https://github.com/mozilla/bedrock/issues/11538, which is in progress
